src/Nancy/Diagnostics/DiagnosticsHook.cs

```diff
@@ -188,7 +188,7 @@ private static Response ExecuteDiagnostics(NancyContext ctx, IRouteResolver rout
 
             if (resolveResult.After != null)
             {
-                resolveResult.After.Invoke(ctx, CancellationToken);
+                resolveResult.After.Invoke(ctx, CancellationToken).Wait();

```

Comment:
Yeah I know and I had a chat with @damianh about it.. the real alternative would be to convert the diagnostics stuff completely to async as well I guess?